ansible_inventory_server/jujurest.py

```diff
@@ -23,25 +23,33 @@
 from ansible_inventory_server import settings
 
 
-async def juju_status(credentials, model_uuid, filters=None):
-    """Deserialize the current Juju status."""
-    with open(settings.CACERT_PATH, 'r') as f:
-        cacert = f.read()
-
-    current_model = Model()
+async def juju_status(parameters):
+    """Returns Juju status, using the specified connection parameters.
+    Returns None on error"""
+    cacert = parameters.get('juju', {}).get('cacert')
+    if cacert is None:
+        with open(settings.CACERT_PATH, 'r') as fin:
+            cacert = fin.read()
+
+    model = Model()
     try:
-        await current_model.connect(
-            uuid=model_uuid if model_uuid else settings.MODEL_UUID,
-            endpoint=settings.CONTROLLER_ENDPOINT,
-            username=credentials.username,
-            password=credentials.password,
-            cacert=cacert)
+        await model.connect(
+            cacert=cacert,
+            username=parameters['juju']['username'],
+            password=parameters['juju']['password'],
+            uuid=parameters.get('juju', {}).get(
+                'model_uuid', settings.MODEL_UUID),
+            endpoint=parameters.get('juju', {}).get(
+                'endpoint', settings.JUJU_ENDPOINT)
+        )
+        status = await model.get_status(
+            parameters.get('juju', {}).get('filters'))
+
```

Comment:
please add `await model.disconnect()` as it was missing

Reply:
used `asyncio.ensure_future()` instead. We do not have to wait for model to disconnect to finish processing the request